src/vellum/workflows/nodes/core/inline_subworkflow_node/node.py

```diff
@@ -36,6 +37,8 @@ def run(self) -> Iterator[BaseOutput]:
             )
             subworkflow_stream = subworkflow.stream(
                 inputs=self._compile_subworkflow_inputs(),
+                event_filter=all_workflow_event_filter,
```

Comment:
Some of the changes in this PR were unrelated and were just context related features I noticed while implementing. Will split these out to merge through in the meantime to keep this PR more focused